.ci/travis_build_opencv.sh

```diff
@@ -23,15 +34,13 @@ if [[ ! -e "$HOME/usr/installed-${OPENCV_VERSION}" ]]; then
         -D BUILD_opencv_python=OFF \
         -D BUILD_opencv_python2=OFF \
         -D BUILD_opencv_python3=OFF \
-        -D CMAKE_INSTALL_PREFIX=$HOME/usr \
+        -D CMAKE_INSTALL_PREFIX=$INSTALL_PATH \
+        -D CMAKE_BUILD_TYPE=Release \
+        -D OPENCV_EXTRA_MODULES_PATH=$OPENCV_CONTRIB \
         -D CUDA_ARCH_BIN=5.2 \
         -D CUDA_ARCH_PTX="" \
         ..
-    make -j4
-    make install && touch $HOME/usr/installed-${OPENCV_VERSION}
-    cd ../..
+    make install && touch $INSTALL_FLAG
+    popd
     touch $HOME/fresh-cache
 fi
-
-sudo cp -r $HOME/usr/include/* /usr/local/include/
```

Comment:
You should copy the headers to `/usr/local/include/`, otherwise, `cargo build` cannot find the header.

Reply:
That's weird. Why it couldn't fetch libs from `usr/include`, but can from `usr/local/include`? 
I suppose we need to change the script the way that it doesn't copy any files. Why not place it in right place at generation time?

Reply:
The default place on most user systems (such as my Mac/Linux) is `/usr/local/{include,lib}`. So that's why the `build.rs` uses those default locations. On Travis, because I wanted to have the library cached, so the install path is first `$HOME/usr` and then the files are copied to the default locations.

This is extra copy, I agree. But I cannot think of a better way that satisfies the cache and the default location at the same time.

Reply:
Edit: here is the default cmake install_prefix,
https://github.com/opencv/opencv/blob/master/CMakeLists.txt#L44

Reply:
Hmm, I see. Why not then user `/usr/local` as prefix for `cmake`?

Reply:
For caching reasons... caching the whole `/usr/local/` seems bold...

Reply:
@nebgnahz but you are caching whole `$HOME/usr`. So it caches  `/usr/local/` and some more...

Reply:
I am only caching `$HOME/usr` which is `/home/travis/usr`, not `/usr/local`. 

`/usr/local/` might have /usr/local/lib/libyaml.a` for example.

Reply:
Hmm, I see. There is some confusion between `$home/usr` and `usr` folders...

Reply:
In my PR, #52, all build passes, except test_cascade_lenna. So it seems OpenCV with contrib returns a different result for cascade detection.

Reply:
Souns weird because this test is actually unchanged...

Reply:
I am equally mystified...

On my linux, this returns
```
[(Rect { x: 217, y: 202, width: 172, height: 172 }, 0)]
```

Reply:
For what is worth, #52 have passed... (I have to increase the tolerance range).
If you can take a look and you feel ok with it, I am planning to merge it.

Reply:
Debugging OpenCV build from scratch is painful... I have to be asynchronous: submit a PR change and wait for the build (around one hour).

Reply:
Yep, I do. I'm going to install WSL tomorrow and try to reproduce it locally. As you can see all my builds takes ~40 min or something. So I'm really a guy who fully understands this pain :D

Reply:
Can I merge #52? What do you think?

Reply:
@nebgnahz sure, I'l just rebase on it :)

Reply:
Hmm, now it's not required: I've merged your branch so my build is not passing :) Thank you.

Now i'm going to make it happen on AppVeyor: choco package doesn't provide opencv_module functionallity.